java/ql/src/Security/CWE/CWE-502/UnsafeDeserialization.ql

```diff
@@ -54,6 +62,33 @@ class UnsafeDeserializationConfig extends TaintTracking::Configuration {
       ma.getArgument(0) = node.asExpr() and
       exists(SafeJsonIoConfig sji | sji.hasFlowToExpr(ma.getArgument(1)))
     )
+    or
+    exists(MethodAccess ma |
+      (
+        ma.getMethod() instanceof FlexjsonDeserializeMethod or
+        ma.getMethod() instanceof JoddJsonParseMethod
+      ) and
+      node.asExpr() = ma.getAnArgument() and
+      (
+        ma.getArgument(1)
+            .(TypeLiteral)
+            .getType()
+            .(ParameterizedType)
+            .getATypeArgument()
+            .(Class)
+            .isFinal() // jodd.json.JsonParser parser = new JsonParser(); parser.allowClass("com.example.*"); parser.setClassMetadataName("class"); User obj = parser.parse(json, User.class);
+        or
+        exists(
+          MethodAccess dma // Specified class type
+        |
+          dma.getMethod() instanceof DeserializerUseMethod and
+          (
+            ma.getQualifier() = dma or
```

Comment:
Worth noting that
```
JSONDeserializer<Person> jsd = new flexjson.JSONDeserializer<Person>();
jsd.use(null, Person.class);
jsd.deserialize(json);
```
is also safe. Without taking on all the cost of a subsidiary Configuration, it might be best to achieve this by modelling all the `use(...)` methods as having value-flow from qualifier to return value, then using `localFlow(any(MethodAccess ma | ma.getMethod() instanceof DeserializerUseMethod).getQualifier(), ...)`

This would be best achieved by something like adding a lib file that is imported from ExternalFlow.qll that contains:

```
class FlexJsonModels extends SummaryModelCsv {
  override predicate row(string row) {
    row =
      [
        "flexjson;JSONDeserializer;false;use;;;Argument[-1];ReturnValue;value"
      ]
  }
}
```

This says that the `use` methods are fluent (i.e., they return `this`), and enables `localFlow` to do the rest of the work for us.

Reply:
Great. This looks like a nice solution and I will learn a new technique from your final version. Thanks.

Reply:
@smowton - for this one, will you make the change or do you want me to make the change following your code snippet? Please advise. Thanks.

Reply:
I'll pick it up from here

Reply:
Thanks @smowton:-)